src/jsx.d.ts

```diff
@@ -14,6 +14,10 @@ export namespace JSXInternal {
 		? Defaultize<Props, Defaults>
 		: Props;
 
+	interface IntrinsicAttributes {
+		key?: any;
```

Comment:
Can a key be anything? I would have expected it to be `string|number`. Not sure the internal methods that use key.

Reply:
We don't have any restrictions in X because we just do a referentially equality check. So it can be indeed anything in our case.